src/createNumbersGenerator.js

```diff
@@ -26,7 +22,21 @@
  * @return {number}
  */
 function createNumbersGenerator(numbers, min, max) {
-  // write code here
+  const [numArr, minNum, maxNum] = [numbers, min, max];
+
+  return () => {
+    if (numArr === undefined) {
+      return undefined;
+    } else if (minNum === undefined && maxNum === undefined) {
+      return numArr.shift();
+    }
+
+    for (let i = 0; i < numArr.length; i++) { // Чому верхній
+      if (numArr[i] >= minNum && numArr[i] <= maxNum) {
```

Comment:
good, but if we will receive only `max` or `min`?

you can solve this problem with another check or add default values to min and max

Reply:
i fixed this problem